app/models/talentlms/facade.rb

```diff
@@ -0,0 +1,50 @@
+###
+# Copyright 2016 - 2020 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/master/LICENSE.md
+###
+
+module Talentlms
+  class Facade
+
+    def initialize
+      @api = Config.first
+    end
+
+    # Login user to TalentLMS
+    #
+    # @@param user [User] the user
+    # @return [String] URL to redirect user to
+    def login(user)
+      login = Login.find_by(user: user)
+      if login.nil?
+        result = create_account(user) if login.nil?
+      else
+        result = @api.post('userlogin', {login: login.login, password: password})
+      end
+      result['login_key']
+    end
+
+    # Create an account in TalentLMS for a user
+    #
+    # @param user [User] the user
+    # @return [String] URL to redirect the user to to login
+    def create_account(user)
+      login = "user_#{user.id}"
+      password = SecureRandom.hex(8)
+      server_domain = ENV['HOSTNAME']
+      account = {
+        first_name: user.first_name,
+        last_name: user.last_name,
+        email: "#{login}@#{server_domain}",
```

Comment:
Is this used elsewhere?  Why not just send user.email?

Reply:
There were two reasons -- Email addresses have to be unique in TalentLMS, so I wanted to make sure we could have multiple accounts across dev/staging/prod, and I wanted to make sure users can't easily get password resets :-)

Reply:
Sounds good, as long as they don't also need to be routable. (Am I reading it correctly that this would generate something like `user_1@hmis.boston.gov`?

Reply:
They shouldn't need to be routable -- the `restrict_email: 'on'` should keep it from generating anything other than things like password reset emails.

That said, it occurs to me that the logins need to be unique too, and conflicts are possible with that... When I had to do this before, we added a host unique offset... I can do that here too... But that will add an ENV to staging/prod...

Reply:
Can you just encode the hostname in the username as well?  Or the RAILS_ENV?